adapters/handlers/rest/dynamic_user/handlers_dynamic_users.go

```diff
@@ -211,8 +214,14 @@ func (h *dynUserHandler) deleteUser(params users.DeleteUserParams, principal *mo
 	if h.isRootUser(params.UserID) {
 		return users.NewDeleteUserUnprocessableEntity().WithPayload(cerrors.ErrPayloadFromSingleErr(errors.New("cannot delete root user")))
 	}
-
-	roles, err := h.dynamicUser.GetRolesForUser(params.UserID)
+	existingUsers, err := h.dynamicUser.GetUsers(params.UserID)
+	if err != nil {
+		return users.NewDeleteUserInternalServerError().WithPayload(cerrors.ErrPayloadFromSingleErr(err))
+	}
+	if len(existingUsers) == 0 {
+		return users.NewDeleteUserNotFound()
+	}
```

Comment:
Should we allow this delete to be a noop if the user does not exist? I.e., return no content here

Reply:
I would like to distinguish the two cases in the clients :) Then we can return True/False there and users can ignore what is returned if they don't case